### DISCUSSION TITLE
Images not working on certain parts

### DISCUSSION THREAD

**Discussion by SamuelAierizer** [solidusio/solidus#5852] [Troubleshooting] [answered]
I'm having issues with inconsistent image loading using Google Cloud Storage in production. I have other applications where it worked fine so I thought surely I messed something up, but once I visited https://shop.robgranito.com/ and saw that the images don't work on similar conditions as for myself, I thought I'd report it. 

I upload an image via the admin interface, it shows up on the admin page 100% correctly but on the front end side sometimes it shows and sometimes it throws some 400 or 404 error and fails to load.

**Solidus Version**:
gem 'solidus', '~> 4.3'

**To Reproduce**

Deploy the new storefront with Google Cloud Storage

Upload images

Notice that on the storefront it's not working. 
- Example would be the "Collections" section or the products page where it sometimes works, and sometimes not and the products show page.

**Current behavior**

Images return 400 or 404 error when you can clearly see it's there on the bucket and the admin page loads the image perfectly 100% of the times.

**Expected behavior**

Images working on the storefront.

**Screenshots**
<img width="1570" alt="Screenshot 2024-09-09 at 21 45 57" src="https://github.com/user-attachments/assets/3e0efa63-e3aa-46e3-8ae5-eba460f0c7da">

<img width="1570" alt="Screenshot 2024-09-09 at 21 47 12" src="https://github.com/user-attachments/assets/6758d808-ed70-48a4-a471-605a169848d0">

<img width="1415" alt="Screenshot 2024-09-09 at 21 48 51" src="https://github.com/user-attachments/assets/fa092cb9-0b5f-45ca-b5fa-e5ada9d635fa">

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Brave

**Comment by jarednorman** [comment, marked as answer]
Are you using ActiveStorage? Solidus (by default) delegates all the image URL generation and file handling to Solidus. The most common issue that people have is that they have configured ActiveStorage such that it creates expiring URLs, rather than static public URLs. Because Solidus has fragment caching on the frontend, those expiring URLs are cached, work until they expire, then stop working. I don't know anything about Google Cloud Storage, so I can offer any specific advice about configuring that backend.

I would confirm that this is the issue. If it is, you can either remove the caching or (preferably) configure the ActiveStorage provider so that it creates public URLs.

**Comment by cdavaz** [comment]
Had this problem as well and after a lot of digging (didn't happen upon this thread unfortunately), found that I needed to mark my service as `public` in `config/storage.yml`:

```yaml
local:
  service: Disk
  root: <%= Rails.root.join("storage") %>
  public: true
```

Note that this works for my application because these are just product images that should be public. If you have other images that you don't want public, this won't work for you.
